formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufTaggedDecoder.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.protobuf.internal
+
+import kotlinx.serialization.*
+import kotlinx.serialization.protobuf.*
+
+internal abstract class ProtobufTaggedDecoder : TaggedBase(), Decoder, CompositeDecoder {
```

Comment:
Now two classes have two completely separated responsibilities: protobuf encoding/decoding and stack manipulation. They are short, nice and concise. Merging them will make them unmanageable (and likely unreadable because of `inner` modifier) mess